File: 프로그래머스/LV2/기능개발.py
def solution(progresses, speeds):
    answer = []
    day = 0
    count = 0
    while len(progresses) > 0:
        if (progresses[0] + day * speeds[0]) >= 100:
            progresses.pop(0)
            speeds.pop(0)
            count += 1
        else:
            if count > 0:
                answer.append(count)
                count = 0
            else:
                day += 1
    answer.append(count)
    return answer

File: 프로그래머스/javascript/LV2/기능개발.js
function solution(progresses, speeds) {
    // 각 기능이 완료되기까지 걸리는 일수 계산
    const days = progresses.map((p, i) => Math.ceil((100 - p) / speeds[i]));

    const answer = [];
    let max = 0;   
    let count = 0; 

    for (const day of days) {
        // 현재 기능이 기준 일수보다 오래 걸리면 → 새 배포 묶음 시작
        if (day > max) {
            if (count > 0) answer.push(count); 
            max = day; 
            count = 0; 
        }
        count++; 
    }

    answer.push(count);
    
    return answer;
}
